src/coreclr/src/vm/gccover.cpp

```diff
@@ -1400,7 +1400,15 @@ BOOL OnGcCoverageInterrupt(PCONTEXT regs)
     }
 
     Thread* pThread = GetThread();
-    _ASSERTE(pThread);
+    if (!pThread)
+    {
+        // No thread at the moment so we aren't doing coverage for this function.
+        // This should only occur for methods with the NativeCallableAttribute,
+        // where the caller could be coming from a thread unknown to the CLR and
+        // we haven't created a thread yet - see PreStubWorker_Premptive().
+        _ASSERTE(pMD->HasNativeCallableAttribute());
+        return (FALSE);
```

Comment:
Does returning FALSE here make the test to work?

I expect we are going to hit `CONSISTENCY_CHECK_MSG(!ExecutionManager::IsManagedCode(GetIP(pContext)), "hit privileged instruction!");` right after the `OnGcCoverageInterrupt` call.

I think this needs to do `RemoveGcCoverageInterrupt` and `return TRUE;`.

Reply:
Hmmm. Let me see what I did wrong then. I see what you are saying.

Reply:
You're right. I was running the test under RET not CHK. I see the error now.